pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -84,6 +85,17 @@ func add(ctx context.Context, mgr manager.Manager, r reconcile.Reconciler) error
 	}
 
 	predBuildRun := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
```

Comment:
I am thinking if we should use the similar way for `build` controller:
Check in the `CreateFunc` if the build is registered, then we should skip this reconciliation?

Reply:
Yeah, I was asking that myself as well. In my opinion we have another conceptual problem in the build controller: the build controller validates the existence of the build strategy and the secrets associated with the build and sets the registered status accordingly. To do this in the correct way, it would need to reconcile all builds in a namespace when any secret is deleted in that namespace because potentially it now has an unresolved secret reference. So, assuming that we agree that this logic is necessary, then we should open an issue for it, but this then also mean that one has to reconcile all builds on controller start.

Reply:
agree. thx!